language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -178,6 +179,13 @@ lookupDynamicBuiltinName dynName = do
             term = Builtin () $ DynBuiltinName () dynName
         Just mean -> pure mean
 
+-- | Instantiate all the free variables of a term by looking them up in an environment.
+dischargeVarEnv :: VarEnv uni -> WithMemory Term uni -> WithMemory Term uni
+dischargeVarEnv varEnv =
+    termSubstFreeNames $ \name -> do
```

Comment:
I recall that the CEK machine allows name shadowing/non-global renaming. This is mentioned in the module header, but perhaps deserves a standalone note.

Reply:
I'd expect people studying the code to read the module header. What's the point of duplicating the same note twice in a single file? Especially when the note is already in the "prime space".

Reply:
Well, in this instance I thought: "hm, does the CEK machine assume globally renamed variables?". Grepped for "renamed", nothing, no note references, eventually I looked in the header.

If you just want to put it in the header then maybe reference it from here.

Reply:
I'll do that, but why here? Global uniqueness is as related here as in all other places in this module.

Reply:
Added a separate note.

Reply:
I guess the question of where the note is relevant is something of a matter of taste. Here it seems to me that it's definitely relevant, but I agree it's somewhat relevant in a lot of places...